http://dev.mutt.org/hg/mutt/rev/05cec973f0d9
http://dev.mutt.org/hg/mutt/rev/8ec606f6cafb

# HG changeset patch
# User Brendan Cully <brendan@kublai.com>
# Date 1268089247 28800
# Node ID 05cec973f0d97065709f2ec55a42b39797995eff
# Parent  817c60d5da7dae3376a6ad683091ae24d7cf2ed2
When regexp compilation fails, report actual pattern along with error.

diff -r 817c60d5da7d -r 05cec973f0d9 pattern.c
--- a/pattern.c	Mon Mar 08 14:58:36 2010 -0800
+++ b/pattern.c	Mon Mar 08 15:00:47 2010 -0800
@@ -254,6 +254,7 @@
 static int eat_regexp (pattern_t *pat, BUFFER *s, BUFFER *err)
 {
   BUFFER buf;
+  char errmsg[STRING];
   int r;
 
   memset (&buf, 0, sizeof (buf));
@@ -290,14 +291,15 @@
   {
     pat->p.rx = safe_malloc (sizeof (regex_t));
     r = REGCOMP (pat->p.rx, buf.data, REG_NEWLINE | REG_NOSUB | mutt_which_case (buf.data));
-    FREE (&buf.data);
     if (r)
     {
-      regerror (r, pat->p.rx, err->data, err->dsize);
-      regfree (pat->p.rx);
+      regerror (r, pat->p.rx, errmsg, sizeof (errmsg));
+      mutt_buffer_printf (err, "'%s': %s", buf.data, errmsg);
+      FREE (&buf.data);
       FREE (&pat->p.rx);
       return (-1);
     }
+    FREE (&buf.data);
   }
 
   return 0;
